be/src/util/table_metrics.cpp

```diff
@@ -40,24 +39,78 @@ void TableMetrics::uninstall(MetricRegistry* registry) {
     UNREGISTER_TABLE_METRIC(load_rows);
 }
 
-void TableMetricsManager::cleanup() {
+bool TableMetricsManager::can_install_metrics() {
+    int64_t old_num = 0;
+    do {
+        old_num = _installed_metrics_num.load();
```

Comment:
I think we should install the metrics of the top scan/load tables which are dynamic, rather than the first registered tables which are relatively static.  If select the top, the implementation might be complex. perhaps we can consider it later.

Reply:
> I think we should install the metrics of the top scan/load tables which are dynamic, rather than the first registered tables which are relatively static. If select the top, the implementation might be complex. perhaps we can consider it later.

@banmoy  good question. I have considered this before. Metrics is a scenario where writes far outweigh reads. The cost of maintaining the top is very high. I doubt the value of doing this in the kernel. I prefer to leave it to an external system.